cclib/parser/gamessdatparser.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Parser for GAMESS(US) .dat output files"""
+
+
+import re
+import numpy
+
+from cclib.parser import logfileparser
+from cclib.parser import utils
+
+
+class GAMESSDAT(logfileparser.Logfile):
+    """A GAMESS .dat log file"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(logname="GAMESS .dat", *args, **kwargs)
+
+    def __str__(self):
+        """Return a string representation of the object."""
+        return f"GAMESS .dat log file {self.filename}"
+
+    def __repr__(self):
+        """Return a representation of the object."""
+        return f'GAMESS .dat ("{self.filename}")'
+    
+    def normalisesym(self, label):
+        """Normalise the symmetries used by GAMESS .dat."""
+
+        pass
+
+    def before_parsing(self):
+        pass
+
+    def extract(self, inputfile, line):
+        """Extract information from the file object inputfile."""
+        
+        # Extract element and its properties
+
+        # $DATA  
+        # water                                                                           
+        # C1       0
+        # O           8.0      0.0000000000      0.0000000000      0.0000000000
+        #    STO     3
+                
+        # H           1.0      0.9900000000      0.0000000000      0.0000000000
+        #    STO     3
+                
+        # H           1.0     -0.2728810000      0.9516490000      0.0000000000
+        #    STO     3
+                
+        #  $END   
+
+        # Extract molecule name
+        name_pattern = r"\$DATA\s+([\w\s]+)"
+        name_match = re.search(name_pattern, line)
+        if name_match:
+            name = name_match.group(1).split('\n')[0].strip()
+            self.metadata["name"] = name
+
+        # Extract atomic information using regex
+        atom_pattern = r"(\w+)\s+([\d.]+)\s+([-+]?\d+\.\d+)\s+([-+]?\d+\.\d+)\s+([-+]?\d+\.\d+)"
+        atom_matches = re.findall(atom_pattern, line)
+
+        atomic_data = []
+        for atom_match in atom_matches:
+            atom_info = {
+                "symbol": atom_match[0],
+                "mass": float(atom_match[1]),
+                "X": float(atom_match[2]),
+                "Y": float(atom_match[3]),
+                "Z": float(atom_match[4])
+            }
```

Comment:
X,Y, Z here will be able to populate `self.atomcoords`

Reply:
Should I keep only the coords then? What about the symbol and mass?